src/Pos/Ssc/GodTossing/Types/Message.hs

```diff
@@ -22,22 +24,47 @@ import           Pos.Ssc.GodTossing.Core       (InnerSharesMap, Opening, SignedC
 import           Pos.Ssc.GodTossing.Toss.Types (GtTag (..))
 import           Pos.Types                     (StakeholderId)
 
--- | Data message. Can be used to send actual data.
-data GtMsgContents
-    = MCCommitment !SignedCommitment
-    | MCOpening !StakeholderId !Opening
-    | MCShares !StakeholderId !InnerSharesMap
-    | MCVssCertificate !VssCertificate
+class HasGtTag a where
+    toGtTag :: a -> GtTag
+
+data MCCommitment = MCCommitment !SignedCommitment
+    deriving (Show, Eq, Generic)
+
+data MCOpening = MCOpening !StakeholderId !Opening
+    deriving (Show, Eq, Generic)
+
+data MCShares = MCShares !StakeholderId !InnerSharesMap
+    deriving (Show, Eq, Generic)
+
+data MCVssCertificate = MCVssCertificate !VssCertificate
     deriving (Show, Eq, Generic)
 
-makePrisms ''GtMsgContents
+makePrisms ''MCCommitment
+makePrisms ''MCOpening
+makePrisms ''MCShares
+makePrisms ''MCVssCertificate
+
+instance Buildable MCCommitment where
+    build _ = "commitment contents"
+
+instance Buildable MCOpening where
+    build _ = "opening contents"
+
+instance Buildable MCShares where
+    build _ = "shares contents"
+
+instance Buildable MCVssCertificate where
+    build _ = "VSS certificate contents"
```

Comment:
Are we sure we don't want to see more information here? I think it would be good to display `StakeholderId` at least.